### PR TITLE
controllers specs

### PR DESCRIPTION
**Pull request by amon-cofie**
### Changes:

- Created request specs for all the controllers
- For each action, I checked:
  - If response status was correct.
  - If a correct template was rendered.
  - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  describe 'GET /users/:id' do
+    before(:each) { get '/users/1' }
+
+    it 'should have status :success' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'should render :show' do
+      expect(response).to render_template(:show)
+    end
```
Comment (GzimAsani) [inline spec/controllers/users_controller_spec.rb]: - [ ] I am getting 4 errors, can you please double check them :pray:

Comment (GzimAsani) [reply spec/controllers/users_controller_spec.rb]: - [ ]  users_spec.rb and posts_spec.rb needs to be on the spec/requests/ folder, kindly update/create that folder. This way is better/easier to check and manage your tests

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @amon-cofie h                        :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your tests, 95% of them ar passing :+1: 

 ## Required Changes 
- [ ] Check the comments below :arrow_double_down: 


 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (amon-cofie) [reply spec/controllers/users_controller_spec.rb]: Thank you for the suggestions. I will make the changes. However, regarding the tests that are not passing, I am not getting any errors on my end and I don't know how to visualize the errors. Perhaps, if I could get more context on this issue, I could be properly oriented toward ways I could fix it. Screenshot of my tests passing: 
image

**Review by mwanawabangona — APPROVED** [review]
Hi @amon-cofie 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**